pkg/distributor/distributor.go

```diff
@@ -233,6 +235,13 @@ func (d *Distributor) Push(ctx context.Context, grpcReq *connect.Request[pushv1.
 	return d.PushParsed(ctx, req)
 }
 
+func (d *Distributor) GetProfileLanguage(series *distributormodel.ProfileSeries) string {
+	if len(series.Samples) == 0 {
+		return "unknown"
+	}
+	return pprof.GetLanguage(series.Samples[0].Profile, d.logger)
```

Comment:
Once the SDKs starts sending the language and more information such as build information this can be adapted to read from the request and fallback to the regex matchers.